_includes/menu.html

```diff
@@ -4,79 +4,49 @@
 manage your navigation system
 {% endcomment %}
 
-<header class="usa-header usa-header--extended" role="banner">
-  <div class="usa-navbar">
-    {% include "logo.html" %}
-    <button type="button" class="usa-menu-btn">Menu</button>
-  </div>
-  <nav
-    id="menu-navigation"
-    aria-label="Menu navigation"
-    role="navigation"
-    class="usa-nav"
-  >
-    <div class="usa-nav__inner">
+<header class="usa-header usa-header--basic usa-header--basic-megamenu" role="banner">
+  <div class="usa-nav-container">
+    <div class="usa-navbar">
+      {% include "logo.html" %}
+      <button type="button" class="usa-menu-btn">Menu</button>
+    </div>
+
+    <nav
+      class="usa-nav"
+      aria-label="Menu navigation"
+    >
       <button type="button" class="usa-nav__close">
         {% image "./node_modules/@uswds/uswds/dist/img/usa-icons/close.svg" "close" %}
       </button>
       <ul class="usa-nav__primary usa-accordion">
         {% for nav_item in primary_navigation %}
-          {% unless nav_item.children %}
-          {% if tags %}
-            {% assign basedir = page.url | remove: titles_roots[tags].root | split: '/' | first | lstrip %}
-            {% assign linkdir = nav_item.url | remove: titles_roots[tags].root | replace: '/', '' | lstrip %}
-          {% else %}
-            {% assign basedir = page.url | remove_first: '/' | split: '/' | first | lstrip %}
-            {% assign linkdir = nav_item.url | replace: "/", "" | lstrip %}
-          {% endif %}
         <li class="usa-nav__primary-item">
+          {% assign nav_item_class = "" %}
```

Comment:
**concern**: This is a lot of mutable state to manage for rendering a header menu.

**idea for a change**: What would you think about including style information in the navigation data file, like under a `class` key? Then rendering would look like, this, only current-page computation needed.

```liquid
{% for link in primary_navigation %}
  {% if link.url == page.url %}
    {{ assign current = "usa-current" }}
  {% endif %}
  <a class="{{ link.class }} {{ current }}" href="{{ link.url | url }}">
  ...
```

Reply:
Refactored this and added contact classes to the navigation data file.